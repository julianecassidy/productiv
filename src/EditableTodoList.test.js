import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

const TEST_TODO_0 = {
  title: "test todo priority 2",
  description: "test desc",
  priority: 2,
  id: "testId",
};

const TEST_TODO_1 = {
  title: "test todo priority 1",
  description: "test desc",
  priority: 1,
  id: "testId",
};

const TEST_TODO_2 = {
  title: "test todo priority 1 2nd version",
  description: "test desc",
  priority: 1,
  id: "testId",
};

const TEST_TODO_3 = {
  title: "test todo priority 3",
  description: "test desc",
  priority: 3,
  id: "testId",
};

function testUpdate() {
  console.log("Called test update!");
}

function testRemove() {
  console.log("Called test remove!");
}

testRemove = jest.fn();
testUpdate = jest.fn();

// toDos, update, remove

const TEST_TODOS = [TEST_TODO_0, TEST_TODO_1, TEST_TODO_2, TEST_TODO_3];

describe("EditableTodoList component", function () {
  it("renders without crashing", function () {
    render(
      <EditableTodoList
        toDos={TEST_TODOS}
        update={testUpdate}
        remove={testRemove}
      />
    );
  });
  it("matches snapshot", function () {
    const { container } = render(
      <EditableTodoList
        toDos={TEST_TODOS}
        update={testUpdate}
        remove={testRemove}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("Renders every editable Todo on the page", function () {
    const { container } = render(
      <EditableTodoList
        toDos={TEST_TODOS}
        update={testUpdate}
        remove={testRemove}
      />
    );

    const testIds = container.querySelectorAll("#testId");
    expect(testIds.length).toEqual(4);
  });
});
