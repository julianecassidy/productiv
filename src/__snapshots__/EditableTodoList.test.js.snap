// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList component matches snapshot 1`] = `
<div>
  <div
    class="toDos"
  >
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="testId"
        >
          <div>
            <b>
              test todo priority 2
            </b>
             
            <small>
              Priority: 
              2
            </small>
          </div>
          <div>
            <small>
              test desc
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="testId"
        >
          <div>
            <b>
              test todo priority 1
            </b>
             
            <small>
              Priority: 
              1
            </small>
          </div>
          <div>
            <small>
              test desc
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="testId"
        >
          <div>
            <b>
              test todo priority 1 2nd version
            </b>
             
            <small>
              Priority: 
              1
            </small>
          </div>
          <div>
            <small>
              test desc
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="testId"
        >
          <div>
            <b>
              test todo priority 3
            </b>
             
            <small>
              Priority: 
              3
            </small>
          </div>
          <div>
            <small>
              test desc
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
